pkg/kube/result.go

```diff
@@ -83,5 +83,5 @@ func (r Result) Intersect(rs Result) Result {
 
 // isMatchingInfo returns true if infos match on Name and GroupVersionKind.
 func isMatchingInfo(a, b *resource.Info) bool {
-	return a.Name == b.Name && a.Mapping.GroupVersionKind == b.Mapping.GroupVersionKind
+	return a.Name == b.Name && a.Mapping.GroupVersionKind.Kind == b.Mapping.GroupVersionKind.Kind
```

Comment:
Shouldn't we change this to `a.Mapping.GroupVersionKind.GroupKind() == b.Mapping.GroupVersionKind.GroupKind()` then? We still want to ensure that two resources that are the same "Kind" (e.g. "Deployment") come from the same "Group" (e.g. "k8s.io"). For example, aggregated APIs like OpenShift could come up with their own Deployment object that acts differently from Kubernetes' Deployment object.

Reply:
No, we can't. Otherwise Deployment in ‘extensions/v1beta1’ won't be matching Deployment in ‘apps/v1beta1’, which is the whole point of this PR